.github/workflows/deploy-preview.yml

```diff
@@ -1,6 +1,6 @@
 name: Deploy PR Preview
 on:
-  pull_request:
+  pull_request_target:
```

Comment:
[This article](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/) seems to suggest that `pull_request_target` is inappropriate for workflows that build the PR branch’s contents, since it has access to the main repository and its secrets. (I don’t know what those secrets are; they could be trivial or they could leave you vulnerable.) The article suggests a different trigger, `workflow_run`, for building untrusted code using the `pull_request` trigger as a trampoline.

Reply:
Using the `workflow_run` trigger would not work in this case because it doesn't have access to the PR number, which is required for the preview upload.

Reply:
To answer the other question, these are the only secrets in the repo:

![image](https://github.com/ZeLonewolf/openstreetmap-americana/assets/3254090/7a749cc0-dba5-49f8-a33a-a9d412f998da)

I have access controls implemented on the AWS side to restrict these credentials to only be able to write to a specific sub-folder in that one s3 button. The risk is a rogue PR that modifies the workflow to somehow log the secrets. But, we'd see the PR and what's in it, and the only thing they'd be able to do is (briefly) upload files to the bucket.

Reply:
The article is clear that, aside from those secrets, the action has access to this repository’s read-write token, which can be used to write to the main branch. We would at least need to set `persist-credentials` to `false`. But if the reason for this unsafe trigger is just to get the PR number, then the article provides a safer approach; see the `ReceivePR.yml` code snippet, which saves the PR number as an artifact.

Reply:
Passing the PR number should work in theory

Reply:
Ah, I see how this works.  The `workflow_run` job only runs in the main branch, not in the PR branch. So we will not able able to test this until merged unfortunately, but it will be secure.